mesonbuild/build.py

```diff
@@ -1247,24 +1247,37 @@ def process_kwargs(self, kwargs):
             raise InvalidArguments(f'Invalid rust_dependency_map "{rust_dependency_map}": must be a dictionary with string values.')
         self.rust_dependency_map = rust_dependency_map
 
-    def _extract_pic_pie(self, kwargs: T.Dict[str, T.Any], arg: str, option: str) -> bool:
+    def _extract_opt_or_arg(self, kwargs: T.Dict[str, T.Any], arg: str, option: str, allow_none: bool = True) -> T.Optional[bool]:
         # Check if we have -fPIC, -fpic, -fPIE, or -fpie in cflags
         all_flags = self.extra_args['c'] + self.extra_args['cpp']
         if '-f' + arg.lower() in all_flags or '-f' + arg.upper() in all_flags:
             mlog.warning(f"Use the '{arg}' kwarg instead of passing '-f{arg}' manually to {self.name!r}")
             return True
 
-        k = OptionKey(option)
+        val: T.Any
```

Comment:
shouldn't this be `val: T.Union[bool, coredata.UserFeatureOption]`?

Reply:
I rewrote the code so that it is `val: T.Optional[bool]` instead.